apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.html.ts

```diff
@@ -0,0 +1,759 @@
+            aria-label="Locked"
+          >
+            <i class="fas fa-lock" aria-hidden="true"></i>
+          </a>
+        `
+      : html`
+          <a href="${urlPrefix}/instance_question/${instance_question.id}/">${rowLabelText}</a>
+        `}
+    ${instance_question.file_count > 0
+      ? html`
+          <a
+            tabindex="0"
+            class="btn btn-xs border text-secondary ml-1"
+            role="button"
+            data-toggle="popover"
+            data-trigger="focus"
+            data-container="body"
+            data-html="true"
+            data-content="Personal notes: ${instance_question.file_count}"
+            aria-label="Has personal note attachments"
+          >
+            <i class="fas fa-paperclip"></i>
+          </a>
+        `
+      : ''}
+  `;
+}
+
+function ExamQuestionHelpBestSubmission() {
+  return html`
+    <a
+      tabindex="0"
+      role="button"
+      data-toggle="popover"
+      data-trigger="focus"
+      data-container="body"
+      data-html="true"
+      data-placement="auto"
+      title="Best submission"
+      data-content="The percentage score of the best submitted answer, or whether the question is <strong>unanswered</strong>, has a <strong>saved</strong> but ungraded answer, or is in <strong>grading</strong>."
+    >
+      <i class="far fa-question-circle" aria-hidden="true"></i>
+    </a>
+  `;
+}
+
+function ExamQuestionHelpAvailablePoints() {
+  return html`
+    <a
+      tabindex="0"
+      role="button"
+      data-toggle="popover"
+      data-trigger="focus"
+      data-container="body"
+      data-html="true"
+      data-placement="auto"
+      title="Available points"
+      data-content="The number of points that would be earned for a 100% correct answer on the next attempt. If retries are available for the question then a list of further points is shown, where the <i>n</i>-th value is the number of points that would be earned for a 100% correct answer on the <i>n</i>-th attempt."
+    >
+      <i class="far fa-question-circle" aria-hidden="true"></i>
+    </a>
+  `;
+}
+
+function ExamQuestionHelpAwardedPoints() {
+  return html`
+    <a
+      tabindex="0"
+      role="button"
+      data-toggle="popover"
+      data-trigger="focus"
+      data-container="body"
+      data-html="true"
+      data-placement="auto"
+      title="Awarded points"
+      data-content="The number of points already earned, as a fraction of the maximum possible points for the question."
+    >
+      <i class="far fa-question-circle" aria-hidden="true"></i>
+    </a>
+  `;
+}
+
+function ConfirmFinishModal({
+  assessment_instance,
+  csrfToken,
+}: {
+  assessment_instance: any;
+  csrfToken: string;
+}) {
+  return Modal({
+    id: 'confirmFinishModal',
+    title: 'All done?',
+    body: html`
+      ${assessment_instance.all_questions_answered
+        ? html` <div class="alert alert-warning">There are still unanswered questions.</div> `
+        : ''}
```

Comment:
This warning is not showing, though it's also not showing on master. I just checked, and this is never set anywhere in the code. It seems to have been introduced in c9a9b2f17256f7f052784e51f9a2a669cb93df39 without an equivalent update in the variables to account for it, so this is a long-term bug. Do you want to remove this code or implement the check somehow?

Reply:
Good catch. I opted to fix this functionality in 6d651df.